src/store/modules/topics.store.js

```diff
@@ -32,56 +30,53 @@ const state = {
 
 const getters = {
     isDefaultTopic: (state) => {
-        return state.current && state.current.id === 'ech'
+        return state.current === 'ech'
     },
     /** Returns the current topic's id, or `ech` if no topic is selected */
     currentTopicId: (state) => {
-        if (state.current) {
-            return state.current.id
-        }
-        return 'ech'
+        return state.current
+    },
+    currentTopic: (state) => {
+        return state.config.find((topic) => topic.id === state.current)
     },
 }
 
 const actions = {
-    setTopics: ({ commit }, topics) => {
-        if (Array.isArray(topics)) {
-            commit('setTopicConfig', topics)
-        }
-    },
-    setTopicTree: ({ commit }, tree) => {
-        if (Array.isArray(tree)) {
-            commit('setTopicTree', tree)
-        }
+    setTopics: ({ commit }, { topics, dispatcher }) => {
+        commit('setTopics', { topics, dispatcher })
     },
-    changeTopic: ({ commit }, topic) => {
-        commit(CHANGE_TOPIC_MUTATION, topic)
+    setTopicTree: ({ commit }, { layers, dispatcher }) => {
+        commit('setTopicTree', { layers: layers.map((layer) => layer.clone()), dispatcher })
     },
-    setTopicById: ({ commit, state }, topicId) => {
-        const topic = state.config.find((topic) => topic.id === topicId)
-        if (topic) {
-            commit(CHANGE_TOPIC_MUTATION, topic)
+    changeTopic: ({ commit, state }, { value, dispatcher }) => {
+        if (
+            state.config.find((topic) => topic.id === value) ||
```

Comment:
to have a boolean response, you can instead use `.some(filter)`

Reply:
done